profiles/rawls-service-perimeters/terraform/rawls-service-perimeters/vpc-flow-egress-alert.tf

```diff
@@ -0,0 +1,157 @@
+  # possible that we want to change them all together, use a static variables here instead of define this as input.
+  sumologic_egress_thresholds = {
+    alert_low_60sec_50mib = {
+      egress_threshold_mib = 50
+      egress_window_sec    = 60
+      cron_expression      = "0 * * * * ? *"
+      time_range           = "-5m"
+      schedule_type        = "RealTime"
+    }
+    alert_mid_10min_150mib = {
+      egress_threshold_mib = 150
+      egress_window_sec    = 600
+      cron_expression      = "0 * * * * ? *"
+      time_range           = "-5m"
+      schedule_type        = "RealTime"
+    }
+    alert_high_60min_200mib= {
+      egress_threshold_mib = 200
+      egress_window_sec    = 3600
+      cron_expression      = "0 0 * * * ? *"
+      time_range           = "-12m"
+      schedule_type        = "1Hour"
+    }
+  }
+
+  // build a map with that contains all egress configurations.
+  vpc_flow_alert_configs = { for entry in setproduct(keys(var.vpc_flow_egress_alerts), keys(local.sumologic_egress_thresholds)) :
+  "${entry[1]}_${entry[0]}" => merge(lookup(var.vpc_flow_egress_alerts, entry[0]), lookup(local.sumologic_egress_thresholds, entry[1]))
+  }
+
+  queries_rendered = { for egress_rule, alert_config in local.vpc_flow_alert_configs :
+  tostring(egress_rule) => templatefile(local.query_path, {
+    aou_env                           = lookup(alert_config, "aou_env", "")
+    egress_threshold_mib              = lookup(alert_config, "egress_threshold_mib", "")
+    egress_window_sec                 = lookup(alert_config, "egress_window_sec", "")
+    sumologic_source_category_name    = lookup(alert_config, "sumologic_source_category_name", "")
+  })
+  }
+
+  queries_encoded = { for egress_rule, query_text in local.queries_rendered :
+  egress_rule => jsonencode(query_text)
+  }
+
+  # Build a map of rendered Content templates for use in the sumologic_content resource and
+  # module outputs
+  egress_rule_to_config = { for egress_rule, alert_config in local.vpc_flow_alert_configs :
+  egress_rule => templatefile(local.content_template_path, {
+    aou_env            = lookup(alert_config, "aou_env", "")
+    webhook_id            = lookup(alert_config, "sumologic_webhook_id_hexadecimal", "")
+    egress_threshold_mib = lookup(alert_config, "egress_threshold_mib", 0)
+    egress_window_sec    = lookup(alert_config, "egress_window_sec", 0)
+    cron_expression      = lookup(alert_config, "cron_expression", 0)
+    schedule_type        = lookup(alert_config, "schedule_type", 0)
+    sumologic_source_category_name        = lookup(alert_config, "sumologic_source_category_name", 0)
+    time_range           = lookup(alert_config, "time_range", 0)
+    query_text           = lookup(local.queries_encoded, egress_rule)
+  })
+  }
+}
+
+# Pull Sumologic credentials from Vault
+data "vault_generic_secret" "sumologic-secret-path" {
+  path = "secret/dsde/firecloud/dev/aou/sumologic"
```

Comment:
IIRC these credentials are not specific to dev? So maybe we put them under `firecloud/common` instead?

Reply:
what is the full path for that? I choose that path because that is the only path that AoU developers has writer permission on.

Reply:
Something like `secret/dsde/firecloud/common/aou/sumologic`? I can add that folder and corresponding permissions. We can make that change later though, don't let it hold up this PR if you don't have permissions to move the secret there now.

Reply:
Would be great if you can help setting up that folder and permission now. Thanks!

Reply:
OK I put the secret into `secret/dsde/firecloud/common/aou/sumologic`. Will push the change in the path up.